### PR TITLE
Update spec file and external deps to use newer celery version

### PR DESCRIPTION
**Pull request by goosemania**
closes #2589
https://pulp.plan.io/issues/2589

### CODE REVIEW

File: deps/external_deps.json [hunk 1]
```diff
@@ -40,14 +40,9 @@
         "platform": ["el6", "el7", "fc23", "fc24"]
     },
     {
-        "name": "python-billiard",
-        "version": "3.3.0.17-2",
-        "platform": ["el6", "el7"]
```
Comment (goosemania) [inline deps/external_deps.json]: python-billiard >= 3.3.0.19 is required for celery 3.1.17.
3.3.0.20 is in EPEL7.

Comment (bmbouter) [inline deps/external_deps.json]: This is great, ty.

File: pulp.spec [hunk 2]
```diff
@@ -389,6 +388,12 @@ Requires: Django14
 %else
 Requires: python-django >= 1.4.0
 %endif
+# RHEL6 or RHEL7
+%if 0%{?rhel} == 6 || 0%{?rhel} == 7
+Requires: python-celery == 3.1.17
+%else
+Requires: python-celery >= 3.1.12
```
Comment (dkliban) [inline pulp.spec]: is 3.1.12 new enough on fedora?

Comment (goosemania) [inline pulp.spec]: Any version >= 3.1.12 will solve bootsteps issue, that's my understanding.
On Fedora celery 3.1.20 is currently available.

Comment (bmbouter) [inline pulp.spec]: What about replacing L#391 - L#396 with `Requires: python-celery >= 3.1.17`?

Comment (goosemania) [inline pulp.spec]: That's a good idea! It does not allow someone to use celery 3.1.12 - 3.1.16 but the probability that it is needed is very low I guess.

**Comment by bmbouter** [comment]
Thanks Tanya!

**Pull request merged**